inc/class-sitemaps.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Class file for the Core_Sitemaps class.
+ * This is the main class integrating all other classes.
+ *
+ * @package Core_Sitemaps
+ */
+
+/**
+ * Class Core_Sitemaps
+ */
+class Core_Sitemaps {
+	/**
+	 * List of registered sitemap providers.
+	 *
+	 * @var Core_Sitemaps_Provider[]
+	 */
+	protected $providers;
+	/**
+	 * Core_Sitemaps constructor.
+	 * Register the registry and bootstrap registered providers.
+	 *
+	 * @uses apply_filters
+	 */
+	public function __construct() {
+		$registry = new Core_Sitemaps_Registry();
+		/**
+		 * Provides a 'core_sitemaps_register_providers' filter which contains a associated array of
+		 * Core_Sitemap_Provider instances to register, with the key passed into it's bootstrap($key) function.
+		 */
+		$this->providers = apply_filters(
+			'core_sitemaps_register_providers',
+			[
+				'index' => new Core_Sitemaps_Index(),
```

Comment:
I don't think the sitemap index should be in this filterable list. It doesn't seem to make sense to allow disabling the sitemap index.

Reply:
Yeah this highlights really that in our implementation the index is implemented as a fake post-type but it isn't really, due to the amount of shared code.

Perhaps we should not have the index extend  the Core_Sitemaps_Provider class at all either, this would clarify the code in other places as well @kirstyburgoine thoughts?

Reply:
Fixed with https://github.com/GoogleChromeLabs/wp-sitemaps/pull/42/commits/4336bd320fd2842f639a67284e7f28aa55caeb44

Reply:
I don't like how `Core_Sitemaps_Provider` focuses only on post types either 🙂 I'll create an issue to describe some changes I'd like to see there.